### DISCUSSION TITLE
Does signOut() requires a delay before being invoked?

### DISCUSSION THREAD

**Discussion by yanickrochon** [nextauthjs/next-auth#5164] [Help]
### Question 💬

We use this module in a standalone authentication app controlling the auth state of other applications within an ecosystem. So, when a user hits any of these apps, a query is made to the authentication app which returns an accessToken to make API requests.

When signing out from one of these apps, a redirection is made to the authentication app and the requested URL calls `signOut` with a redirect to the sign-in page.

However, it does not work.

I tried having this React component :

```js
const AutoLogout = ({ callbackUrl }) => {
   signOut({ callbackUrl });

   return null;
};
```

But the user session is not destroyed. Then I tried :

```js
const AutoLogout = ({ callbackUrl }) => {
   useEffect(() => {
      signOut({ callbackUrl });
   }, [callbackUrl]);

   return null;
}
```
But the user session is not destroyed. Perhaps the session has not loaded yet? So I tried this :

```js
const AutoLogout = ({ callbackUrl }) => {
   const { status } = useSession();

   useEffect(() => {
      if (status !== 'loading') {
         signOut({ callbackUrl });
      }
   }, [status, callbackUrl]);

   return null;
}
```

Still, the user session is not destroyed! Perhaps there is a need to wait a little longer? 

```js
const AutoLogout = ({ callbackUrl }) => {
   const { status } = useSession();

   useEffect(() => {
      if (status !== 'loading') {
         setTimeout(() => signOut({ callbackUrl }), 200);
      }
   }, [status, callbackUrl]);

   return null;
}
```

And it **finally** worked! So I am left clueless as to why such a delay is required before calling the function.

### How to reproduce ☕️

See the code provided above.

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by BrianUribe6** [comment]
I'm currently facing this issue as well. I have a page `/auth/logout` that does something of this sort:

```ts
function LogOut() {
  useEffect(() => {
   signOut({callbackUrl: "/"});
  }, []);
 return <p>Signing out...</p>;
}
```

I have tried wrapping signOut on an async function and awaiting it inside the useEffect. I have tried the same approach, but also passing
`redirect: false` and then redirecting using `router.push("/")`, and the only solution that worked was adding a delay on the time out.

Why is this delay necessary?
